Content.Client/LateJoin/LateJoinGui.cs

```diff
@@ -25,10 +25,13 @@ public sealed class LateJoinGui : SS14Window
         [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
         [Dependency] private readonly IClientConsoleHost _consoleHost = default!;
 
-        public event Action<string>? SelectedId;
+        public event Action<(uint, string)> SelectedId;
 
-        private readonly Dictionary<string, JobButton> _jobButtons = new();
-        private readonly Dictionary<string, BoxContainer> _jobCategories = new();
+        private readonly Dictionary<uint, Dictionary<string, JobButton>> _jobButtons = new();
+        private readonly Dictionary<uint, Dictionary<string, BoxContainer>> _jobCategories = new();
+        private readonly List<ScrollContainer> _jobLists = new();
+
+        private readonly Control _base;
 
         public LateJoinGui()
```

Comment:
Should probably be XAML'd, but I guess that can be done seperately

Reply:
UI has very little business as XAML, all the XAML would contain is an empty BoxContainer.